internal/operators/manager.go

```diff
@@ -0,0 +1,159 @@
+	GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string
+}
+
+// NewManager creates new instance of an Operator Manager
+func NewManager(log logrus.FieldLogger, hostApi host.API) Manager {
+	cfg := ocs.Config{}
+	err := envconfig.Process("myapp", &cfg)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	return NewManagerWithConfig(log, hostApi, &cfg)
+}
+
+// NewManagerWithConfig creates new instance of an Operator Manager
+func NewManagerWithConfig(log logrus.FieldLogger, hostApi host.API, cfg *ocs.Config) Manager {
+	ocsValidator := ocs.NewOCSValidator(log.WithField("pkg", "ocs-operator-state"), hostApi, cfg)
+	return Manager{
+		log:                log,
+		ocsValidatorConfig: cfg,
+		ocsValidator:       ocsValidator,
+	}
+}
+
+// GenerateManifests generates manifests for all enabled operators.
+// Returns map assigning manifest content to its desired file name
+func (mgr *Manager) GenerateManifests(cluster *common.Cluster) (map[string]string, error) {
+	lsoEnabled := false
+	ocsEnabled := mgr.checkOCSEnabled(cluster)
+	if ocsEnabled {
+		lsoEnabled = true // if OCS is enabled, LSO must be enabled by default
+	} else {
+		lsoEnabled = mgr.checkLSOEnabled(cluster)
+	}
+	operatorManifests := make(map[string]string)
+
+	if lsoEnabled {
+		manifests, err := mgr.generateLSOManifests(cluster)
+		if err != nil {
+			mgr.log.Error("Cannot generate LSO manifests due to ", err)
+			return nil, err
+		}
+		for k, v := range manifests {
+			operatorManifests[k] = v
+		}
+	}
+
+	if ocsEnabled {
+		manifests, err := mgr.generateOCSManifests(cluster)
+		if err != nil {
+			mgr.log.Error("Cannot generate OCS manifests due to ", err)
+			return nil, err
+		}
+		for k, v := range manifests {
+			operatorManifests[k] = v
+		}
+	}
+	return operatorManifests, nil
+}
+
+// AnyOperatorEnabled checks whether any operator has been enabled for the given cluster
+func (mgr *Manager) AnyOperatorEnabled(cluster *common.Cluster) bool {
+	return mgr.checkLSOEnabled(cluster) || mgr.checkOCSEnabled(cluster)
+}
+
+// ValidateOCSRequirements validates OCS requirements. Returns "true" if OCS operator is not deployed
+func (mgr *Manager) ValidateOCSRequirements(cluster *common.Cluster) bool {
+	if isEnabled(cluster, models.OperatorTypeOcs) {
+		return mgr.ocsValidator.ValidateOCSRequirements(&cluster.Cluster)
+	}
+	return true
+}
+
+// GetOperatorStatus gets status of an operator of given type.
+func (mgr *Manager) GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string {
+	operator, err := findOperator(cluster, operatorType)
+	if err != nil {
+		return "Something went wrong with Unmarshalling operators"
+	}
+	if operator != nil {
+		if swag.BoolValue(operator.Enabled) {
+			return operator.Status
+		}
+	}
+	return "OCS is disabled"
+}
+
+func (mgr *Manager) generateLSOManifests(cluster *common.Cluster) (map[string]string, error) {
+	mgr.log.Info("Creating LSO Manifests")
```

Comment:
Logging is inconsistent. We only log information about generating either LSO or OCS manifests.

Reply:
What kind of consistency do you have in mind? There is no generic interface there yet so the code is strongly LSO- and OCS- centric.

Reply:
I see only 2 methods log I wonder whether we should add logging to more or move both to corresponding `Manifests` method.

Reply:
When the whole feature is implemented, these two methods should be gone and at that point the logging will be in a operator-type-generic `GenerateManifests`. Are you OK with having that changed later?

Reply:
Yes